app/forms/common/Time.scala

```diff
@@ -20,50 +20,60 @@ import java.text.DecimalFormat
 import java.time.LocalTime
 import java.time.format.DateTimeFormatter
 
-import play.api.data.Forms.{optional, text}
+import play.api.data.Forms.text
 import play.api.data.{Forms, Mapping}
 import play.api.libs.json.{Json, OFormat}
+import utils.validators.forms.FieldValidator._
 
 import scala.util.Try
 
-case class Time(time: LocalTime) {
-
-  private val inputFormat = DateTimeFormatter.ofPattern("HH:mm")
-
-  def toInputFormat: String = time.format(inputFormat)
-
-  override def toString: String = time.toString
-
-}
+case class Time(time: LocalTime)
 
 object Time {
   implicit val format: OFormat[Time] = Json.format[Time]
 
   val hourKey = "hour"
   val minuteKey = "minute"
+  val ampmKey = "ampm"
+
+  val time12HourFormatter = DateTimeFormatter.ofPattern("h:mma")
+
+  val am = "AM"
+  val pm = "PM"
 
   val mapping: Mapping[Time] = {
-    def build(hour: Try[Int], minutes: Try[Int]): Try[LocalTime] =
+    def build(hour: Try[Int], minutes: Try[Int], ampm: String): Try[LocalTime] =
       for {
         h <- hour
         m <- minutes
-      } yield LocalTime.of(h, m)
+      } yield {
+        val timeString = s"$h:${f"$m%02d"}$ampm"
+        LocalTime.parse(timeString, time12HourFormatter)
+      }
 
-    def bind(hour: Try[Int], minutes: Try[Int]): Time =
-      build(hour, minutes)
+    def bind(hour: Try[Int], minutes: Try[Int], ampm: String): Time =
+      build(hour, minutes, ampm)
         .map(apply)
         .getOrElse(throw new IllegalArgumentException("Could not build time - missing one of parameters"))
 
-    def unbind(time: Time): (Try[Int], Try[Int]) =
-      (Try(time.time.getHour), Try(time.time.getMinute))
+    def unbind(time: Time): (Try[Int], Try[Int], String) = {
+      val timeString = time.time.format(time12HourFormatter)
+      (Try(timeString.split(":")(0).toInt), Try(time.time.getMinute), timeString.takeRight(2))
```

Comment:
There is method `getHour`. Is it possible to use it on the first argument?

For second one you used `getMinute`.

Additionally I found method `get` in the `LocalTime` https://docs.oracle.com/javase/8/docs/api/java/time/LocalTime.html#get-java.time.temporal.TemporalField-
There you can specify what element from the time you want to retrieve. There is link to supported fields
https://docs.oracle.com/javase/8/docs/api/java/time/LocalTime.html#isSupported-java.time.temporal.TemporalField-
Some of the fields that could be useful here:
- HOUR_OF_AMPM
- CLOCK_HOUR_OF_AMPM (Not sure what's this, this will require some testing around)
- MINUTE_OF_HOUR
- AMPM_OF_DAY

Reply:
I was looking for something like that - getting the hour as a number from 1-12 and getting the AM/PM indictor.  In the end I just turned the time into a 12-hour format string, as you can see.   I'll take another look.